wgengine/magicsock/magicsock.go

```diff
@@ -866,6 +870,17 @@ func (c *Conn) SetBlockEndpoints(block bool) {
 		c.endpointsUpdateActive = true
 		go c.updateEndpoints(why)
 	}
+
+	// Clear any endpoints from the peerMap if block is true.
+	if block {
+		c.peerMap.forEachEndpoint(func(ep *endpoint) {
+			// Simulate receiving a call-me-maybe disco packet with no
+			// endpoints, which will cause all endpoints to be removed.
```

Comment:
This doesn't actually remove all endpoints, just the ones previously discovered by CallMeMaybe.  Is there a reason we don't want to just blow them all away?

I feel like we've got a lot of piecemeal tweaks that block endpoints from different sources.  Wouldn't it be cleaner just to block them all at the level of `de.endpointState`?

Reply:
I think this is fine. Magicsock's "block endpoints" just means "don't send local endpoints to peers via disco and don't accept endpoints from peers via disco". The comment on the option is written this way.

The only other way peers can arrive is if they're written via coordination, which is also written in the comment on the option (and already blocked in coder's tailnet package).

Reply:
Actually, I might look into just wiping out all endpoints rather than only the call-me-maybe ones. It would be nice to get rid of all the blockEndpoints plumbing in the coder tailnet package in favor of a simple implementation.

Reply:
Done